compiler/injectdestructors.nim

```diff
@@ -625,7 +626,7 @@ template handleNestedTempl(n, processCall: untyped, willProduceStmt = false,
         #Condition needs to be destroyed outside of the condition/branch scope
         branch[0] = p(it[0], c, s, normal)
 
-      branch[^1] = if it[^1].typ.isEmptyType or willProduceStmt:
+      branch[^1] = if it[^1].typ.isEmptyType or willProduceStmt or withoutElse:
```

Comment:
But `it[^1].typ.isEmptyType` should really be true. Other parts in the compiler use the same check!

Reply:
Let me try again with setting its type to empty in the sem phase. I failed last since I didn't change the logic of `maybeVoid`.

Reply:
If `it[^1].typ.isEmptyType` is true, the destructors for its return is not called. Though I could probably make an exception for implicit discardable return type.

```nim
proc main() =
  if false:
    newNode()
```